contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -213,27 +280,48 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
 
         (address recipient, uint256 amount) = abi.decode(message, (address, uint256));
         require(recipient != address(0), "NativeTokenDestination: zero recipient address");
-        require(amount != 0, "NativeTokenDestination: zero transfer value");
+
+        uint256 scaledAmount = _scaleTokens(amount, true);
 
         // If the contract has not yet been collateralized, we will deduct as many tokens
         // as needed from the transfer as needed. If there are any excess tokens, they will
         // be minted and sent to the recipient.
-        uint256 adjustedAmount = amount;
-        if (currentReserveImbalance > 0) {
-            if (amount > currentReserveImbalance) {
-                emit CollateralAdded({amount: currentReserveImbalance, remaining: 0});
-                adjustedAmount = amount - currentReserveImbalance;
+        uint256 adjustedAmount = scaledAmount;
+        uint256 reserveImbalance = currentReserveImbalance;
+        if (reserveImbalance > 0) {
+            if (scaledAmount > reserveImbalance) {
+                emit CollateralAdded({amount: reserveImbalance, remaining: 0});
+                adjustedAmount = scaledAmount - reserveImbalance;
```

Comment:
since `adjustedAmount=scaledAmount` I think we should just use `adjustedAmount -= reserveImbalance` in this `if/else` block, and not use `scaledAmount`. Easier to logic through imo

Reply:
another option is in the line you declare `uint256 adjustedAmount = scaledAmount;` just make it `uint256 adjustedAmount;`, since you assign teh variable values in both `if/else` anyways

Reply:
Oh yes I like option 2, thanks!